src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBwaAlignerSpark.java

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemIndexSingleton;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Wrapper class for using the PathSeq Bwa aligner class in Spark. Encapsulates closing the index when done.
+ */
+public final class PSBwaAlignerSpark implements AutoCloseable {
+
+    private final JavaSparkContext ctx;
+    private final Broadcast<PSBwaArgumentCollection> bwaArgsBroadcast;
+
+    public PSBwaAlignerSpark(final JavaSparkContext ctx, final PSBwaArgumentCollection bwaArgs) {
+        this.ctx = ctx;
+        this.bwaArgsBroadcast = ctx.broadcast(bwaArgs);
+    }
+
+    private static JavaRDD<GATKRead> doBwaAlignmentHelper(final JavaRDD<GATKRead> reads,
```

Comment:
Capturing ```this``` (and thus forcing it to be serialized) happens when the lambda references a field from ```this```.  Avoid it by copying the field to a temporary final variable.
Example of the bad thing:
```
class Foo {
   private final GiantKitchenSinkFullOfGarbage = new GiantKitchenSinkFullOfGarbage();
   private final int minMapQ = 47;
   JavaRDD<GATKRead> filter( final JavaRDD<GATKRead> reads ) {
        return reads.filter(read -> read.getMappingQuality() >= minMapQ);
    }
}
```
Fixed:
```
class Foo {
   private final GiantKitchenSinkFullOfGarbage = new GiantKitchenSinkFullOfGarbage();
   private final int minMapQ = 47;
   JavaRDD<GATKRead> filter( final JavaRDD<GATKRead> reads ) {
        final int tmpMinMapQ = minMapQ;
        return reads.filter(read -> read.getMappingQuality() >= tmpMinMapQ);
    }
}
```